blocks/init/src/Blocks/components/heading/heading.php

```diff
@@ -26,19 +26,17 @@
 
 $headingClass = Components::classnames([
 	Components::selector($componentClass, $componentClass),
+	Components::selector($blockClass, $blockClass),
 	Components::selector($blockClass, $blockClass, $selectorClass),
 	Components::selector($additionalClass, $additionalClass),
 ]);
 
 $headingLevel = $headingLevel ? "h{$headingLevel}" : 'h2';
 
-$unique = Components::getUnique();
+$unique = $attributes['uniqueWrapperId'] ?? Components::getUnique();
+echo Components::outputCssVariables($attributes, $manifest, $unique, $globalManifest); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
Once the "all in one style" update is done, this won't matter. But didn't we say we will keep the `<style>` inside the component?